src/config.js

```diff
@@ -17,7 +23,7 @@ export class Config {
    * @return {Config}
    */
   registerEndpoint(name, configureMethod, defaults = {}) {
-    let newClient        = new HttpClient();
+    let newClient        = this.httpClientFactory();
```

Comment:
nothing is passed to the factory. so, doesn't serve any point yet.

cf usage of factory: https://github.com/aurelia/dependency-injection/issues/38#issuecomment-117807908